### PR TITLE
brew: remove sudo call only in docker context

### PR DESCRIPTION
**Pull request by iMichka**
Follow up from #6018

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: Library/Homebrew/brew.rb [hunk 1]
```diff
@@ -123,7 +113,14 @@ def create_tap(cmd, tap_name)
 
     # Unset HOMEBREW_HELP to avoid confusing the tap
     ENV.delete("HOMEBREW_HELP") if help_flag
-    create_tap(cmd, possible_tap.name)
+    tap_commands = []
+    cgroup = Utils.popen_read("cat", "/proc/1/cgroup")
+    if !cgroup.include?("azpl_job") && !cgroup.include?("docker")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/brew.rb]: Replaced the`unless .. ||` as `unless` with complex conditionals are hard to parse (for me at least).

**Comment by MikeMcQuaid** [comment]
Thanks @iMichka and sorry for my pedantry!

**Comment by iMichka** [comment]
Thanks @MikeMcQuaid !
